src/DeserializerFactory.php

```diff
@@ -29,45 +29,33 @@ class DeserializerFactory {
 	 */
 	private $legacyFactory;
 
-	/**
-	 * @var CurrentDeserializerFactory
-	 */
-	private $currentFactory;
-
 	public function __construct( Deserializer $dataValueDeserializer, EntityIdParser $idParser ) {
 		$this->legacyFactory = new LegacyDeserializerFactory( $dataValueDeserializer, $idParser );
-		$this->currentFactory = new CurrentDeserializerFactory( $dataValueDeserializer, $idParser );
 	}
 
 	/**
+	 * @param DispatchableDeserializer $currentEntityDeserializer
+	 *
 	 * @return Deserializer
 	 */
-	public function newEntityDeserializer() {
+	public function newEntityDeserializer( DispatchableDeserializer $currentEntityDeserializer ) {
```

Comment:
I'm not objecting this change, at least not it's intention. But this method signature is now really weird and confusing. Why do you have to pass an entity serializer to a method called `newEntitySerializer`? Sounds like somebody suggests to build endless loops. Can we rename this stuff somehow so it becomes more obvious what all this is about?